### DISCUSSION TITLE
Would it be worthwhile to make `sort` stable?

### DISCUSSION THREAD

**Discussion by jrm5100** [samtools/bcftools#1926] [Ideas]
The [cmp_bcf_pos](https://github.com/samtools/bcftools/blob/875c067bdd31b3021e8148fc129dbedebda6208f/vcfsort.c#L97-L119) function looks at:

- id
- pos
- alleles

However, if all of these match the records are considered equivalent and `qsort` is unstable. Theoretically it would be possible to track the original index of the records as they are read and use that as a final tiebreaker.

Is this something that would be worthwhile? This isn't entirely theoretical- I ran into it but I could accept that it shouldn't matter how those records are sorted.

**Comment by pd3** [comment]
We can make this into a feature request, this would be a clean way of doing things.

If anyone wanted to do it, some easy rewrites would be required. Instead of writing out `bcf1_t` structures, they need to be wrapped into a struct with the original index record https://github.com/samtools/bcftools/blob/9b352b7c5431b5cbb33dd5141864c389a1c8e0a0/vcfsort.c#L61
